geps/gep-1619.md

```diff
@@ -159,47 +167,380 @@ Generally, the implementation API programs the dataplane API; however these two
+implementations. However, certain cookie attributes require special consideration, which we will discuss in detail.
+
+##### Path
+
+The `Path` attribute should be automatically determined by implementors. When the `SessionPersistencePolicy` policy is
+attached to an xRoute, the implementor should interpret the path as configured on the xRoute. However, if the
+`SessionPersistencePolicy` policy is attached to a service, the path should be left unset.
+
+To interpret the `Path` attribute from an xRoute, implementors should take note of the following:
+1. For an xRoute that matches all paths, the `Path` should be set to `/`.
+2. For an xRoute that has multiple paths, the `Path` should be interpreted based on the path taken.
+3. For an xRoute using a path that is a regex, the `Path` should be set to the longest non-regex prefix (.e.g. if the
+   path is /p1/p2/*/p3 and the request path was /p1/p2/foo/p3, then the cookie path would be /p1/p2).
+
+##### Secure, HttpOnly, SameSite
+
+The `Secure`, `HttpOnly`, and `SameSite` cookie attributes are security-related. The API implementers should follow the
+security-by-default principle and configure these attributes accordingly. This means enabling `Secure` and `HttpOnly`,
+and setting `SameSite` to `Strict`.
+
+#### Session Persistence API with GAMMA
+
+The object of the [GAMMA (Gateway API for Mesh Management and Administration)](https://gateway-api.sigs.k8s.io/contributing/gamma/)
+initiative is to provide support for service mesh and mesh-adjacent use-cases with Gateway API. Mesh traffic routing with
+Mesh traffic routing with Gateway API is accomplished with `xRoutes` specifying service-to-service traffic opposed to
+gateway-to-service traffic.
+
+Given that service meshes commonly have session persistence requirements, this API design should take into consideration
+session persistence needs in GAMMA and service mesh scenarios.
+
+#### Cookie Origination
+
+Most implementations implement session persistence by originating cookies at gateways. This spec covers the use cases
+related to that particular architecture. However, it's important to note that this spec does not require cookies to
+originate solely at a gateway. The cookie and persistent session could originate from a backend, a sidecar in a backend,
+or any other infrastructure component.
+
+#### Cookie Rewriting
+
+Cookie rewriting happens when the backend generates the cookie with the same name before the gateway has the opportunity
+to create it, and then the gateway modifies the cookie's value and/or attributes. While this scenario was previously
+discussed as a [use case](#Use-Cases-Cookies), during the implementation of this specification, it is up to the
+individual implementations to determine how to handle such cases. An implementation has the freedom to either rewrite
+the cookie or choose not to modify it, resulting in the absence of a persistent session.
+
+```go
```

Comment:
nit: this is under "Cookie Rewriting", should be under "API"?

Reply:
Yea good point.

Moved under a `GO` section as a lot of other GEPs have done.